engine/resources/docker.go

```diff
@@ -0,0 +1,292 @@
+// Mgmt
+// Copyright (C) 2013-2018+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package resources
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+	"github.com/purpleidea/mgmt/engine"
+	"github.com/purpleidea/mgmt/engine/traits"
+)
+
+func init() {
+	engine.RegisterResource("docker", func() engine.Res { return &DockerRes{} })
+}
+
+// DockerRes is a docker account resource.
+type DockerRes struct {
```

Comment:
So since docker has turned into a behemoth, I'm wondering if it's best to do like aws and have various sub resources, like aws:ec2, etc... So we'd have docker:image, or whatever...

Reply:
It makes sense to create new resources to implement different docker features. `docker:network` comes to mind. If we want to create a `docker:image` resource, it would be a matter of pulling out the image search and image pull requests (about six lines of code,) and plumbing in some auto-edges to the new resource. I think it makes sense to get this merged first with the image stuff, and pull it out later if we need to.

Reply:
Just saw this comment now. Sounds good. Shouldn't we name this something different? docker:???

Reply:
Yeah. `docker:container` is boring but descriptive. Any better ideas? Also, do colons work with the language yet?

Reply:
Yes in theory, but not in git master yet, and not tested yet:

https://github.com/purpleidea/mgmt/blob/6405b2371bb7a03a23ab67890fc4530930527f7f/lang/lexer.nex#L362

You can try branch feat/import for new things, but might be broken, eat your dog, get rebased often, etc...